ros/lib/utils.py

```diff
@@ -210,40 +210,62 @@ def get_psi_count(queryset: db.Model, bool_flag: bool) -> int:
+            instance_type_name, None)
         if instance_type_properties is None:
             # logging lookup failure on Prometheus
             ec2_instance_lookup_failures.labels(reporter='API Events').inc()
-            return 'NA'
+    return instance_type_properties
 
+
+def generate_highlight_description(instance_type, cloud_provider, regions):
+    """Returns string that is description against instance type."""
+    instance_type_properties = instance_type_info_by_name(
+        instance_type, cloud_provider)
+    regions_info = ",".join(regions) + ' regions' if regions else ''
+
+    if instance_type_properties is not None:
         cpu_type = instance_type_properties['extra']['physicalProcessor']
         num_vcpus = instance_type_properties['extra']['vcpu']
         ram_gb = instance_type_properties['extra']['memory']
-        cloud_regions = instance_type_properties['extra']['regionCode']
-
-    description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
-                       f'and {ram_gb} of RAM, running on {cloud_provider} ' \
-                       f'{cloud_regions} regions'
+        description_text = f'{cpu_type} instance with {num_vcpus} vCPUs ' \
+                           f'and {ram_gb} of RAM, running on {cloud_provider} ' \
+                           f'{regions_info}'
+        return description_text
 
-    return description_text
+    return 'NA'
 
 
 def highlights_instance_types(queryset, highlight_type):
     instance_candidates, values_dict, highlights_list = [], {}, []
+    regions_by_type = {}
     if highlight_type == 'current':
         for record in queryset:
             try:
                 _instance_type = record.rule_hit_details[0]['details']['instance_type']
             except (IndexError, KeyError):
                 _instance_type = None
-            instance_candidates.append(_instance_type if _instance_type else record.instance_type)
+            curr_instance_type = _instance_type if _instance_type else record.instance_type
+            instance_candidates.append(curr_instance_type)
+            if curr_instance_type not in regions_by_type:
+                regions_by_type[curr_instance_type] = []
+            if record.region not in regions_by_type[curr_instance_type]:
+                regions_by_type[curr_instance_type].append(record.region)
+
     elif highlight_type in ['suggested', 'historical']:
         for _record in queryset:
             try:
-                instance_candidates.append(_record.rule_hit_details[0]['details']['candidates'][0][0])
+                instance_type = _record.rule_hit_details[0]['details']['candidates'][0][0]
+                instance_candidates.append(instance_type)
+                if instance_type not in regions_by_type:
+                    regions_by_type[instance_type] = []
+                if _record.region not in regions_by_type[instance_type]:
+                    regions_by_type[instance_type].append(_record.region)
```

Comment:
Hello @saltgen, please let me know if you still think to keep these two blocks (i.e `current` & `suggested/historical`) separately? OR may I extract out into one method?

Reply:
I feel its out of the context for this pr, but if possible we can separate out for block in a method? I think that's what I am also feeling